src/main.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
```

Comment:
generally speaking - your main script should be "easy to read" in that it doesnt contain a bunch of declarations and functions and stuff. It should just be a list of operations, basically. Assume whoever is reading the code has an IDE and can quickly jump to function definitions for more info. Function titles are already descriptive - someone should be able to get a rough big picture view of what you're doing by reading the main script